ols/app/endpoints/ols.py

```diff
@@ -86,14 +87,20 @@ def conversation_request(
     previous_input = retrieve_previous_input(user_id, llm_request)
 
     # Retrieve attachments from the request
-    retrieve_attachments(llm_request)
+    attachments = retrieve_attachments(llm_request)
 
     # Log incoming request
     logger.info(f"{conversation_id} Incoming request: {llm_request.query}")
 
     # Redact the query
     llm_request = redact_query(conversation_id, llm_request)
 
+    # Redact all attachments
+    attachments = redact_attachments(conversation_id, attachments)
+
+    # All attachments should be appended to query
+    llm_request.query = append_attachments_to_query(llm_request.query, attachments)
```

Comment:
i don't suppose it matters too much either way, but why did you prefer to redact the attachments separately, rather than construct the entire prompt(with the injected attachment content) and then redact the whole thing?

Reply:
well in the future when we store attachments in conversation DB separately, it might be valuable. But yes, no strong opinion about that

Reply:
that's a good reason